viamonvif/gosoap/gosoap.go

```diff
@@ -0,0 +1,352 @@
+// Package gosoap provides a minimal soap client for viamonvif
```

Comment:
this is terrable code which also fails security lint... 

But it is working now and pulling it in as vendored code allows us to improve it in the future.